RecoEgamma/ElectronIdentification/plugins/ElectronMVAEstimatorRun2Spring15NonTrig.cc

```diff
@@ -0,0 +1,463 @@
+  tmpTMVAReader.AddVariable("ele_kfchi2",           &_allMVAVars.kfchi2);
+  tmpTMVAReader.AddVariable("ele_gsfchi2",        &_allMVAVars.gsfchi2);
+
+  // Energy matching
+  tmpTMVAReader.AddVariable("ele_fbrem",           &_allMVAVars.fbrem);
+
+  tmpTMVAReader.AddVariable("ele_gsfhits",         &_allMVAVars.gsfhits);
+  tmpTMVAReader.AddVariable("ele_expected_inner_hits",             &_allMVAVars.expectedMissingInnerHits);
+  tmpTMVAReader.AddVariable("ele_conversionVertexFitProbability",  &_allMVAVars.convVtxFitProbability);
+
+  tmpTMVAReader.AddVariable("ele_ep",              &_allMVAVars.EoP);
+  tmpTMVAReader.AddVariable("ele_eelepout",        &_allMVAVars.eleEoPout);
+  tmpTMVAReader.AddVariable("ele_IoEmIop",         &_allMVAVars.IoEmIoP);
+  
+  // Geometrical matchings
+  tmpTMVAReader.AddVariable("ele_deltaetain",      &_allMVAVars.deta);
+  tmpTMVAReader.AddVariable("ele_deltaphiin",      &_allMVAVars.dphi);
+  tmpTMVAReader.AddVariable("ele_deltaetaseed",    &_allMVAVars.detacalo);
+  
+  // Spectator variables  
+  tmpTMVAReader.AddSpectator("ele_pT",             &_allMVAVars.pt);
+  tmpTMVAReader.AddSpectator("ele_isbarrel",       &_allMVAVars.isBarrel);
+  tmpTMVAReader.AddSpectator("ele_isendcap",       &_allMVAVars.isEndcap);
+  tmpTMVAReader.AddSpectator("scl_eta",            &_allMVAVars.SCeta);
+
+  tmpTMVAReader.AddSpectator("ele_eClass",                 &_allMVAVars.eClass);
+  tmpTMVAReader.AddSpectator("ele_pfRelIso",               &_allMVAVars.pfRelIso);
+  tmpTMVAReader.AddSpectator("ele_expected_inner_hits",    &_allMVAVars.expectedInnerHits);
+  tmpTMVAReader.AddSpectator("ele_vtxconv",                &_allMVAVars.vtxconv);
+  tmpTMVAReader.AddSpectator("mc_event_weight",            &_allMVAVars.mcEventWeight);
+  tmpTMVAReader.AddSpectator("mc_ele_CBmatching_category", &_allMVAVars.mcCBmatchingCategory);
+
+  //
+  // Book the method and set up the weights file
+  //
+  std::unique_ptr<TMVA::IMethod> temp( tmpTMVAReader.BookMVA(_MethodName , weightFile.fullPath() ) );
+
+  return std::unique_ptr<const GBRForest> ( new GBRForest( dynamic_cast<TMVA::MethodBDT*>( tmpTMVAReader.FindMVA(_MethodName) ) ) );
+}
+
+// A function that should work on both pat and reco objects
+std::vector<float> ElectronMVAEstimatorRun2Spring15NonTrig::
+fillMVAVariables(const edm::Ptr<reco::Candidate>& particle,
+                 const edm::Event& iEvent ) const {
+
+  // 
+  // Declare all value maps corresponding to the products we defined earlier
+  //
+  edm::Handle<reco::BeamSpot> theBeamSpot;
+  edm::Handle<reco::ConversionCollection> conversions;
+
+  // Get data needed for conversion rejection
+  iEvent.getByLabel(_beamSpotLabel, theBeamSpot);
+
+  // Conversions in miniAOD and AOD have different names,
+  // but the same type, so we use the same handle with different tokens.
+  iEvent.getByLabel(_conversionsLabelAOD, conversions);
+  if( !conversions.isValid() )
+    iEvent.getByLabel(_conversionsLabelMiniAOD, conversions);
```

Comment:
is the content the same?
If not, since both will be available in a reco workflow,
for reproducibility purposes you'd probably like to read the miniAOD version if the IDs are a part of the miniAOD object creation.